### PR TITLE
cpu/stm32: rewrite clock configuration documentation

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is a complete rewrite of the STM32 clock configuration documentation. It's part of #14975.
With the recent refactoring work on STM32 clock configuration, the current documentation is completely outdated.

### Testing procedure

`make doc` and read (I'll report here when CircleCI doc build is ready)

### Issues/PRs references

Tick an item in #14975

### CODE REVIEW

**Comment by aabadie** [comment]
rebased and adapted after #15192 got merged.

File: cpu/stm32/doc.txt [hunk 1]
```diff
@@ -1,53 +1,96 @@
...
+This module contains the code and definitions for STM32 cpu
+families supported by RIOT: F0, F1, F2, F3, F4, F7, G0, G4, L0, L1, L4, L5 and WB.
+
+All constants described below can be configured using `CFLAGS` from the command
+line or in the application Makefile.
+
+Clock configuration
+===================
+
+All STM32 cpus have clock configuration code and macro. It is recommend to read
```
Comment (benpicco) [inline cpu/stm32/doc.txt:15]:
```suggestion
All STM32 cpus have clock configuration code and macros. It is recommend to read
```

**Review by benpicco — APPROVED** [review]
Looks good, nice overview.

File: cpu/stm32/doc.txt (hunk 1 above)
Comment (aabadie) [reply cpu/stm32/doc.txt:15]: fixed

**Pull request merged**